### PR TITLE
Add unit test for default look

### PR DESCRIPTION
**Pull request by JGamache-autodesk**

### CODE REVIEW

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 1]
```diff
@@ -421,13 +421,22 @@ namespace
             PvtPrim* assignPrim = stage->createPrim(parent->getPath(), RtToken(matAssign->getName()), RtMaterialAssign::typeName());
             RtMaterialAssign rtMatAssign(assignPrim->hnd());
             
-            PvtPrim* collection = findPrimOrThrow(RtToken(matAssign->getCollectionString()), parent);
-            rtMatAssign.getCollection().addTarget(collection->hnd());
+            if (!matAssign->getCollectionString().empty()) {
```
Comment (JGamache-autodesk) [inline source/MaterialXRuntime/RtFileIo.cpp]: String can be empty when reloading a WIP.

File: source/MaterialXRuntime/RtFileIo.cpp [hunk 2]
```diff
@@ -421,13 +421,22 @@ namespace
...
+            }
+
+            if (!matAssign->getMaterial().empty()) {
+                PvtPrim* material = findPrimOrThrow(RtToken(matAssign->getMaterial()), parent);
+                rtMatAssign.getMaterial().addTarget(material->hnd());
+            }
 
-            PvtPrim* material = findPrimOrThrow(RtToken(matAssign->getMaterial()), parent);
-            rtMatAssign.getMaterial().addTarget(material->hnd());
+            if (matAssign->hasAttribute(MaterialAssign::EXCLUSIVE_ATTRIBUTE)) {
```
Comment (JGamache-autodesk) [inline source/MaterialXRuntime/RtFileIo.cpp]: Attribute is optional and will throw if string is empty. Check beforehand for presence of the attribute.

File: resources/LookDev/emptyLook.mtlx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<!-- Test the RtFileIo can read look where all optional elements are missing,
+	 simulating re-reading a file with unconnected Rt nodes. -->
+<materialx version="1.38">
+  <look name="emptyLook">
+  </look>
+  <look name="emptyMatAssign">
+    <materialassign name="defaultMaterialAssign" material=""/>
```
Comment (bernardkwok) [inline resources/LookDev/emptyLook.mtlx]: I think this should fail Document::validate() and throw an exception there. Could be we're not validating on read?
If it's called and it's not failing we should patch this up.

Comment (JGamache-autodesk) [reply resources/LookDev/emptyLook.mtlx]: There is value in being able to re-read a work in progress. 
Is is valid for rendering? Definitely not. 
But it is valid enough for opening in the the node-editor.

Comment (bernardkwok) [reply resources/LookDev/emptyLook.mtlx]: I guess my point is, is this a valid document -- and it is not. A material assignment to nothing has no meaning.

**Pull request merged**